### PR TITLE
Enhance bm_diff

### PR DESCRIPTION
**Pull request by ncteisen**
This fixes some issues bm_diff had with the new metrics from allocs_per_write and atm_add and others. Those were stored as strings in the json file, so changed_ratio would error out. This fixes that and also fixes some issues when the data is written in scientific form.

It also adds a command line flag in case you want to specify which metrics to track, since the output can get pretty messy if you do all of them.

### CODE REVIEW

File: tools/profiling/microbenchmarks/bm_diff.py [hunk 1]
```diff
@@ -86,12 +102,13 @@ def min_change(pct):
...
   details = ''
-  for fld, chk in _INTERESTING:
+  for fld in args.track:
+    chk = _INTERESTING[fld]
     if fld not in n or fld not in o: continue
     if chk(n[fld], o[fld]):
       row.append(changed_ratio(n[fld], o[fld]))
       if details: details += ', '
-      details += '%s:%r-->%r' % (fld, o[fld], n[fld])
+      details += '%s:%r-->%r' % (fld, int(o[fld]), int(n[fld]))
```
Comment (ctiller) [inline tools/profiling/microbenchmarks/bm_diff.py]: Shouldn't this be float() instead of int(): the per-iteration values are floats

Comment (ncteisen) [inline tools/profiling/microbenchmarks/bm_diff.py]: Done

**Comment by ctiller** [comment]
@ncteisen @matt-kwong this and #10101 and #10095 seem to be doing similar things: with #10101 in now, can we recheck and decide amongst ourselves what's still needed?

**Comment by matt-kwong** [comment]
I'll close #10095 in favor for this

**Comment by ncteisen** [comment]
No tests for this, merging early

**Pull request merged**